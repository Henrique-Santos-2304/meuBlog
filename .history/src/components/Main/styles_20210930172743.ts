import styled, { css, keyframes } from "styled-components";

export const Wrapper = styled.main`
  ${({ theme: { colors } }) => css`
    background-color: ${colors.black800};
  `};
  color: #fff;
  width: 100%;
  height: 100vh;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;
const blinkTextCursor = keyframes`
  from {border-right-color: rgba(0, 0, 0, .75);}
  to {border-right-color: transparent}
`;

const textTitle = keyframes`
  from {color: transparent;}
  to {color: #f8f8f8}
`;

export const Tittle = styled.h1`
  animation: ${textTitle} 3s infinite linear;
  ${({ theme: { colors } }) => css`
    color: ${colors.white800};
  `};
  font-size: 5rem;
`;
export const Text = styled.p`
  margin-top: 2rem;
`;

export const TextCursor = styled.span`
  border-right: 2px solid rgba(0, 0, 0, 0.75);
  /* display: inline; */

  /* A m√°gica acontece aqui */
  animation: ${blinkTextCursor} 3s infinite 3s normal;
`;
export const Container = styled.div`
  color: #f00;
  font-size: 3rem;
`;
