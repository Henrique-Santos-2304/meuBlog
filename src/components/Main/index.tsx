/* eslint-disable @next/next/no-img-element */
import { Container, Text } from "@chakra-ui/react";

const Main = () => {
  return (
    <Container
      height="90vh"
      maxW="full"
      bgGradient="linear(to-l,#a8eb12 ,#00bf72 ,#008793 ,#004d7a 45%, #051937)"
    >
      <Text textAlign="center" color="whiteAlpha.800" fontSize="6rem">
        Seção main
      </Text>
    </Container>
  );
};

export default Main;
