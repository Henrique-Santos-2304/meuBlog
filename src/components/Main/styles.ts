import styled, { css } from "styled-components";

export const MyDescription = styled.main`
  text-align: center;
  display: flex;
  flex-direction: column;
  width: 100vw;
  height: 100vh;
`;
export const ContentDescription = styled.div`
  ${({ theme }) => css`
    width: max-content;
    padding: 1rem 2.9rem 1rem 1.4rem;
    margin: 2.4rem 0 0 2rem;
    background: rgba(255, 255, 255, 0.15);
    color: ${theme.colors.secondary};
    display: flex;
    position: relative;
    flex-direction: column;
    align-items: flex-start;
    font-weight: 500;
    border-radius: 2rem;

    &::before {
      content: "";
      position: absolute;
      top: 50%;
      left: 1rem;
      transform: translateY(-50%);
      height: 80%;
      width: 0.3rem;
      background: ${theme.colors.effectsPrimary};
    }
  `}
`;
export const Name = styled.h1.attrs({
  title: "Meu Nome",
})`
  ${({ theme }) => css`
    text-align: start;
    font-size: 2.3rem;
    margin-left: 1.3rem;
    font-weight: 300;
    strong {
      color: ${theme.colors.effectsPrimary};
      font-style: italic;
    }
  `}
`;
export const PhotoUser = styled.img`
  width: 30rem;
  height: 40rem;
  align-self: flex-end;
  justify-self: flex-end;
`;
export const Welcome = styled.p`
  ${({ theme }) => css`
    margin-left: 1.3rem;
    span {
      color: ${theme.colors.effectsPrimary};
    }
  `}
`;

export const Dev = styled.p`
  margin-left: 1.3rem;
`;
