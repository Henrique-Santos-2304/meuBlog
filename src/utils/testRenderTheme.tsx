import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
