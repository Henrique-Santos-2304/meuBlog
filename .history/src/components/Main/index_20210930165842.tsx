import { useEffect, useState } from "react";
import * as S from "./styles";
import Image from "next/image";

const Main = () => {
  const value = "Desenvolvedor Web";
  const [text, setText] = useState("");

  useEffect(() => {
    typeWriter(value);
  }, []);
  const typeWriter = (text: string, i = 0) => {
    if (i < value.length) {
      setText(text.slice(0, i + 1));
      setTimeout(() => {
        typeWriter(text, i + 1);
      }, 100);
    }
  };
  if (text === value) {
    setTimeout(() => {
      setText("");
      typeWriter(value);
    }, 3000);
  }

  return (
    <Image>
      <S.Wrapper>
        <S.Tittle> Henrique dos Santos</S.Tittle>
        <S.Container>
          <S.Text>{text}</S.Text>
          <S.TextCursor />
        </S.Container>
      </S.Wrapper>
    </Image>
  );
};

export default Main;
