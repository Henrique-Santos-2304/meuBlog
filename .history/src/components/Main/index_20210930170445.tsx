import { useEffect, useState } from "react";
import * as S from "./styles";
import imageBackground from "../../../public/img/background.png";

const Main = () => {
  const value = "Desenvolvedor Web";
  const [text, setText] = useState("");

  useEffect(() => {
    typeWriter(value);
  }, []);
  const typeWriter = (text: string, i = 0) => {
    if (i < value.length) {
      setText(text.slice(0, i + 1));
      setTimeout(() => {
        typeWriter(text, i + 1);
      }, 100);
    }
  };
  if (text === value) {
    setTimeout(() => {
      setText("");
      typeWriter(value);
    }, 3000);
  }

  return (
    <S.Wrapper img={imageBackground}>
      <S.Tittle> Henrique dos Santos</S.Tittle>
      <S.Container>
        <S.Text>{text}</S.Text>
        <S.TextCursor />
      </S.Container>
    </S.Wrapper>
  );
};

export default Main;
