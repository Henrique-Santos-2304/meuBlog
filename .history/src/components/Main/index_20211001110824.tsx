import { useCallback, useEffect, useState } from "react";
import * as S from "./styles";

const Main = () => {
  const value = "Desenvolvedor Web";
  const [text, setText] = useState("");

  //eslint-disable
  const typeWriter: any = useCallback(
    (text: string, i = 0) => {
      if (i < value.length) {
        setText(text.slice(0, i + 1));
        setTimeout(() => {
          typeWriter(text, i + 1);
        }, 100);
      }
    },
    [typeWriter]
  );
  if (text === value) {
    setTimeout(() => {
      setText("");
      typeWriter(value);
    }, 3000);
  }
  useEffect(() => {
    typeWriter(value);
  }, [typeWriter]);

  return (
    <S.Wrapper>
      <S.Tittle> Henrique dos Santos</S.Tittle>
      <S.Container>
        <S.Text>{text}</S.Text>
        <S.TextCursor />
      </S.Container>
    </S.Wrapper>
  );
};

export default Main;
