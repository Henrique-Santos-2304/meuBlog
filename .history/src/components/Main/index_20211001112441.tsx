import { useEffect, useState } from "react";
import * as S from "./styles";

const Main = () => {
  const value = "Desenvolvedor Web";
  const [text, setText] = useState("");

  useEffect(() => {
    typeWriter(value);
  }, []);
  const typeWriter = (text: string, i = 0) => {
    if (i < value.length) {
      setText(text.slice(0, i + 1));
      setTimeout(() => {
        typeWriter(text, i + 1);
      }, 100);
    }
  };
  if (text === value) {
    setTimeout(() => {
      setText("");
      typeWriter(value);
    }, 3000);
  }

  return (
    <S.Wrapper>
      <S.Tittle> Henrique dos Santos</S.Tittle>
      <S.Container>
        <S.Text>{text}</S.Text>
        <S.TextCursor />
      </S.Container>
    </S.Wrapper>
  );
};

export default Main;
