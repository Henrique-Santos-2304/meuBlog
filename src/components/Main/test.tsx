import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";
import "jest-styled-components";

import Main from ".";

describe("<Main />", () => {
  it("should render the heading", () => {
    renderTheme(<Main />);

    expect(screen.getByRole("heading", { name: /Oi/i })).toBeInTheDocument();
  });
});
