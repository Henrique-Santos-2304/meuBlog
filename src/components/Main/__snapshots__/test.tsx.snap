// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render  1`] = `
<div>
   
  <div
    class="chakra-container css-8ljg7m"
  >
    <p
      class="chakra-text css-1d3rlnc"
    >
      Seção main
    </p>
  </div>
</div>
`;
